src/software/new_geom/util/intersection.h

```diff
@@ -5,6 +5,23 @@
 #include "software/new_geom/ray.h"
 #include "software/new_geom/segment.h"
 
+/**
+ * Computes the point of intersection between two lines.
+ * Note: this computes the intersection of two lines, not line segments.
+ *
+ * Overlapping lines are considered parallel and not intersecting.
+ *
+ * See:
+ * https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection#Given_two_points_on_each_line
+ *
+ * @params a, b points that represent the first line
+ * @params c, d points that represent the second line
+ *
+ * @return the point of intersection, if it exists
+ */
+std::optional<Point> intersection(const Point &a, const Point &b, const Point &c,
```

Comment:
This was made public for use in `bool intersectsNew(const Ray &first, const Segment &second)`. The other option I saw was to combine `intersection` and `intersects` into a single file, but I would prefer to keep them separate.